gmso/tests/test_improper.py

```diff
@@ -15,17 +15,8 @@ def test_improper_nonparametrized(self):
         site3 = Site(name='site3')
         site4 = Site(name='site4')
 
-        assert site1.n_connections == 0
```

Comment:
This goes for a lot of these tests, but instead of testing a connection of a site, should we instead check that the `connection` object we initialize contains the sites we expect them to have?